### PR TITLE
SRS Server Api (EXPOSUREAPP-14188, 14189)

### PR DESCRIPTION
**Pull request by mtwalli**
## Test 
- Switch to `Mock-Cloud` env
- `Test Menu -> Submission Test Options -> SRS Submission`
- Try submit 
- To disable device time check -> use Switch

## TODO 
- [x] More unit tests 

## Tickets
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14188
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14189

### CODE REVIEW

File: Corona-Warn-App/src/deviceForTesters/res/layout/fragment_test_submission.xml [hunk 1]
```diff
@@ -35,6 +35,44 @@
...
+                android:layout_height="wrap_content"
+                android:layout_marginTop="10dp"
+                android:checked="true"
+                android:text="Check Device Time" />
+
+            <TextView
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginBottom="10dp"
+                android:text="Note: This checkbox has effect only as long as you are in this test menu. it does not have an effect on real submission time check." />
```
Comment (SamuraiKek) [inline Corona-Warn-App/src/deviceForTesters/res/layout/fragment_test_submission.xml:58]: Technically, it's a switch, not a checkbox

Comment (mtwalli) [reply Corona-Warn-App/src/deviceForTesters/res/layout/fragment_test_submission.xml:58]: will leave that to Test menu changes ticket , but you understood  right 😄  ?

Comment (SamuraiKek) [reply Corona-Warn-App/src/deviceForTesters/res/layout/fragment_test_submission.xml:58]: And couldn't we make this switch work outside of the test menu? How would we test the flow when we keep getting `TIME_SINCE_ONBOARDING_UNVERIFIED`?

Comment (mtwalli) [reply Corona-Warn-App/src/deviceForTesters/res/layout/fragment_test_submission.xml:58]: any changes will be addressed by https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14182

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/core/server/SrsAuthorizationServer.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        withContext(dispatcherProvider.IO) {
+            try {
+                authoriseRequest(request)
+            } catch (e: Exception) {
+                throw when (e) {
+                    is SrsSubmissionException -> e
+                    is CwaUnknownHostException,
+                    is NetworkReadTimeoutException,
+                    is NetworkConnectTimeoutException -> SrsSubmissionException(ErrorCode.SRS_OTP_NO_NETWORK, cause = e)
+                    // otherwise blame the server
```
Comment (SamuraiKek) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/core/server/SrsAuthorizationServer.kt:44]: :smile:

**Review by SamuraiKek — APPROVED** [review]
Nice work!

File: Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/submission/ui/SubmissionTestFragmentViewModel.kt [hunk 3]
```diff
@@ -75,6 +79,20 @@ class SubmissionTestFragmentViewModel @AssistedInject constructor(
 
     val tekHistory = MutableLiveData<List<TEKHistoryItem>>()
 
+    fun submit(checkDeviceTime: Boolean) = launch {
+        try {
+            srsSubmissionRepository.submit(SrsSubmissionType.SRS_REGISTERED_RAT, checkDeviceTime = checkDeviceTime)
```
Comment (schauersbergern) [inline Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/submission/ui/SubmissionTestFragmentViewModel.kt:84]: Should this default to SRS_REGISTERED_RAT?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/core/model/SrsDeviceAttestationRequest.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+        if (!scenarioPayload.contentEquals(other.scenarioPayload)) return false
+        if (configData != other.configData) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = scenarioPayload.contentHashCode()
+        result = 31 * result + (configData?.hashCode() ?: 0)
```
Comment (schauersbergern) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/core/model/SrsDeviceAttestationRequest.kt:26]: Is this conversion formula from tec spec?

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/core/model/SrsDeviceAttestationRequest.kt:26]: it is auto generated from Android Studio when data class has `ByteArray` as property

File: Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/submission/ui/SubmissionTestFragmentViewModel.kt (hunk 3 above)
Comment (mtwalli) [reply Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/submission/ui/SubmissionTestFragmentViewModel.kt:84]: it is test menu, any type is fine

**Review by schauersbergern — APPROVED** [review]

**Pull request merged**